gef-ui/server/controllers/ukef-cover-start-date/index.test.js

```diff
@@ -0,0 +1,173 @@
+import { processCoverStartDate } from './index';
+import api from '../../services/api';
+
+const Chance = require('chance');
+
+const chance = new Chance();
+
+jest.mock('../../services/api');
+
+const MockResponse = () => {
+  const res = {};
+  res.redirect = jest.fn();
+  res.render = jest.fn();
+  return res;
+};
+
+const MockRequest = () => {
+  const req = {};
+  req.params = {};
+  req.body = {
+    ukefCoverStartDate: true,
+    day: 1,
+    month: 12,
+    year: 2021,
+  };
+  req.query = {};
+  req.params.applicationId = '1234';
+  req.params.facilityId = '4321';
+  req.success = {
+    message: 'Cover start date for 4321 confirmed',
+  };
+  req.url = '/gef/application-details/1234/confirm-cover-start-date';
+  req.session = {
+    user: {
+      bank: { id: 'BANKID' },
+      roles: ['MAKER'],
+    },
+    userToken: 'TEST',
+  };
+  return req;
+};
+
+const MockApplicationResponse = () => {
+  const res = {};
+  res._id = '1234';
+  res.exporterId = '123';
+  res.bankId = 'BANKID';
+  res.bankInternalRefName = 'Internal refernce';
+  res.additionalRefName = 'Additional reference';
+  res.status = 'DRAFT';
+  res.userId = 'mock-user';
+  res.supportingInformation = {
+    status: 'NOT_STARTED',
+  };
+  res.eligibility = {
+    criteria: [
+      { id: 12, answer: null, text: 'Test' },
+    ],
+    status: 'IN_PROGRESS',
+  };
+  res.editedBy = ['MAKER_CHECKER'];
+  res.submissionType = 'Automatic Inclusion Application';
+  res.submissionCount = 0;
+  res.comments = [];
+  res.ukefDealId = null;
+  res.createdAt = chance.timestamp();
+  res.submissionDate = chance.timestamp();
+  return res;
+};
+
+const MockUserResponse = () => ({
+  username: 'maker',
+  bank: { id: 'BANKID' },
+  firstname: 'Joe',
+  surname: 'Bloggs',
+  timezone: 'Europe/London',
+});
+
+const MockExporterResponse = () => {
+  const res = {};
```

Comment:
could we get these values declared the same way `MockEligibilityCriteriaResponse` is? 
 we could return an object directly, without having to declare the `res` variable

Reply:
Done

Reply:
sorry, I meant in all requests, not just this one :)

Reply:
i.e. `MockRequest` as well, etc